pkg/operator/sync.go

```diff
@@ -113,3 +103,130 @@ func (optr *Operator) waitForDeploymentRollout(resource *appsv1.Deployment) erro
 		return false, nil
 	})
 }
+
+func newDeployment(config *OperatorConfig, features map[string]bool) *appsv1.Deployment {
+	replicas := int32(1)
+	template := newPodTemplateSpec(config, features)
+
+	return &appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "machine-api-controllers",
+			Namespace: config.TargetNamespace,
+			Labels: map[string]string{
+				"api":     "clusterapi",
+				"k8s-app": "controller",
+			},
+		},
+		Spec: appsv1.DeploymentSpec{
+			Replicas: &replicas,
+			Selector: &metav1.LabelSelector{
+				MatchLabels: map[string]string{
+					"api":     "clusterapi",
+					"k8s-app": "controller",
+				},
+			},
+			Template: *template,
+		},
+	}
+}
+
+func newPodTemplateSpec(config *OperatorConfig, features map[string]bool) *corev1.PodTemplateSpec {
+	containers := newContainers(config, features)
+	tolerations := []corev1.Toleration{
+		{
+			Key:    "node-role.kubernetes.io/master",
+			Effect: corev1.TaintEffectNoSchedule,
+		},
+		{
+			Key:      "CriticalAddonsOnly",
+			Operator: corev1.TolerationOpExists,
+		},
+		{
+			Key:               "node.alpha.kubernetes.io/not-ready",
+			Effect:            corev1.TaintEffectNoExecute,
+			Operator:          corev1.TolerationOpExists,
+			TolerationSeconds: pointer.Int64Ptr(120),
+		},
+		{
+			Key:               "node.alpha.kubernetes.io/unreachable",
```

Comment:
shouldn't this be `- key: "node.kubernetes.io/not-ready"`

Reply:
true, thanks